src/python/pants/backend/python/typecheck/mypy/rules.py

```diff
@@ -250,49 +255,60 @@ async def mypy_typecheck_partition(
     )
 
 
-# TODO(#10864): Improve performance, e.g. by leveraging the MyPy cache.
-@rule(desc="Typecheck using MyPy", level=LogLevel.DEBUG)
-async def mypy_typecheck(
+# TODO(#10863): Improve the performance of this, especially by not needing to calculate transitive
+#  targets per field set. Doing that would require changing how we calculate interpreter
+#  constraints to be more like how we determine resolves, i.e. only inspecting the root target
+#  (and later validating the closure is compatible).
+@rule(desc="Determine if necessary to partition MyPy input", level=LogLevel.DEBUG)
+async def mypy_determine_partitions(
```

Comment:
Basically a duplicate of Pylint, minus using the first-party plugins for interpreter constraints.